static/app/views/settings/project/projectFilters/projectFiltersSettings.tsx

```diff
@@ -193,236 +195,236 @@ class LegacyBrowserFilterRow extends Component<RowProps, RowState> {
   }
 }
 
+function CustomFilters({project, disabled}: {disabled: boolean; project: Project}) {
+  return (
+    <Feature
+      features={['projects:custom-inbound-filters']}
+      hookName="feature-disabled:custom-inbound-filters"
+      project={project}
+      renderDisabled={({children, ...props}) => {
+        if (typeof children === 'function') {
+          return children({
+            ...props,
+            renderDisabled: p => (
+              <FeatureDisabled
+                featureName={t('Custom Inbound Filters')}
+                features={p.features}
+                alert={PanelAlert}
+                message={t(
+                  'Release and Error Message filtering are not enabled on your Sentry installation'
+                )}
+              />
+            ),
+          });
+        }
+        return null;
+      }}
+    >
+      {({hasFeature, organization, renderDisabled, ...featureProps}) => (
+        <Fragment>
+          {!hasFeature &&
+            typeof renderDisabled === 'function' &&
+            // XXX: children is set to null as we're doing tricksy things
+            // in the renderDisabled prop a few lines higher.
+            renderDisabled({
+              organization,
+              hasFeature,
+              children: null,
+              ...featureProps,
+            })}
+
+          {customFilterFields.map(field => (
+            <FieldFromConfig
+              key={field.name}
+              field={field}
+              disabled={disabled || !hasFeature}
+            />
+          ))}
+
+          {hasFeature && project.options?.['filters:error_messages'] && (
+            <PanelAlert type="warning" data-test-id="error-message-disclaimer">
+              {t(
+                "Minidumps, errors in the minified production build of React, and Internet Explorer's i18n errors cannot be filtered by message."
+              )}
+            </PanelAlert>
+          )}
+        </Fragment>
+      )}
+    </Feature>
+  );
+}
+
 type Props = {
   features: Set<string>;
-  organization: Organization;
   params: {
     projectId: string;
   };
   project: Project;
 };
 
-type State = {
-  hooksDisabled: ReturnType<(typeof HookStore)['get']>;
-} & AsyncComponent['state'];
+type Filter = {
+  active: boolean;
+  description: string;
+  hello: string;
+  id: string;
+  name: string;
+};
 
-class ProjectFiltersSettings extends AsyncComponent<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      hooksDisabled: HookStore.get('feature-disabled:custom-inbound-filters'),
-    };
-  }
+export function ProjectFiltersSettings({project, params, features}: Props) {
+  const organization = useOrganization();
+  const {projectId} = params;
+
+  const projectEndpoint = `/projects/${organization.slug}/${projectId}/`;
+  const filtersEndpoint = `${projectEndpoint}filters/`;
+
+  const {
+    data: filterList = [],
```

Comment:
i don't think this does anything since the default value is null
```ts
const {x = []} = {x:null}
console.log(x) // null
```

Reply:
oh, you are right...it just made the compiler happy :) will update it...thanks for the heads up

Reply:
typescript tells me that the value can be only `undefined`. Maybe we should update the types to default to `null`

Reply:
oh okay didn't realize it was undefined